dataset/components/version/views.py

```diff
@@ -0,0 +1,154 @@
+# Copyright (C) 2022 Indoc Research
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+
+from aioredis import StrictRedis
+from common import LoggerFactory
+from fastapi import APIRouter
+from fastapi import BackgroundTasks
+from fastapi import Depends
+from fastapi import Response
+
+from dataset.components.dataset.crud import DatasetCRUD
+from dataset.components.dataset.dependencies import get_dataset_crud
+from dataset.components.exceptions import NotFound
+from dataset.components.object_storage.s3 import S3Client
+from dataset.components.parameters import PageParameters
+from dataset.components.parameters import SortParameters
+from dataset.components.schemas import LegacyResponseSchema
+from dataset.components.version.crud import VersionCRUD
+from dataset.components.version.dependencies import get_version_crud
+from dataset.components.version.dependencies import get_version_publisher
+from dataset.components.version.parameters import VersionFilterParameters
+from dataset.components.version.parameters import VersionSortByFields
+from dataset.components.version.publisher import PublishVersion
+from dataset.components.version.schemas import VersionCreateSchema
+from dataset.components.version.schemas import VersionListResponseSchema
+from dataset.config import get_settings
+from dataset.dependencies.redis import get_redis_client
+from dataset.dependencies.s3 import get_s3_client
+
+settings = get_settings()
+logger = LoggerFactory(
+    'api_version',
+    level_default=settings.LOG_LEVEL_DEFAULT,
+    level_file=settings.LOG_LEVEL_FILE,
+    level_stdout=settings.LOG_LEVEL_STDOUT,
+    level_stderr=settings.LOG_LEVEL_STDERR,
+).get_logger()
+router = APIRouter()
```

Comment:
I suggest to define tags on router so it can be avoided in view functions.
```suggestion
router = APIRouter(tags=['version'])
```